pulsar/internal/connection.go

```diff
@@ -764,7 +769,8 @@ func (c *connection) handleSendError(cmdError *pb.CommandError) {
 		errMsg := fmt.Sprintf("server error: %s: %s", cmdError.GetError(), cmdError.GetMessage())
 		request.callback(nil, errors.New(errMsg))
 	case pb.ServerError_TopicTerminatedError:
-		// TODO: no-op
+		errMsg := fmt.Sprintf("server error: %s: %s", cmdError.GetError(), cmdError.GetMessage())
+		c.failPendingRequests(errors.New(errMsg))
```

Comment:
Is this the correct behavior or should it just remove the pending request for the requestID. What if there are other pending requests for a different topic?

Reply:
you are right